src/main/java/com/datastax/oss/protocol/internal/request/AuthResponse.java

```diff
@@ -21,6 +21,12 @@
 import com.datastax.oss.protocol.internal.ProtocolConstants;
 import java.nio.ByteBuffer;
 
+/**
+ * Note that, if the token is writable, the built-in codec will <b>clear its contents</b>
+ * immediately after writing it (to avoid keeping sensitive information in memory). If you want to
+ * reuse the same buffer across multiple message instances, make it {@link
+ * ByteBuffer#asReadOnlyBuffer() read-only}.
```

Comment:
It makes sense to allow read-only because some auth providers might have messages that are reusable and not sensitive (the DSE authenticators are a good example).

Reply:
So the security here is only when an AuthProvider implementation uses read-write buffers. This means it is on the AuthProvider implementation to use a read-write buffer if the information is sensitive, otherwise the info will not be cleared. At least that is how I am understanding it, correct?

Reply:
Exactly. Marking the buffer as read-only is a way to tell that you want to reuse the information.